protocols/Mass_Spec_Prep_2017-12-4/Mass_Spec_Prep_2017-12-4.py

```diff
@@ -0,0 +1,274 @@
+from opentrons import containers, instruments
+from collections import defaultdict
+from otcustomizers import FileInput, StringSelection
+
+"""
+Column A
+"""
+trough = containers.load('trough-12row', 'A1')
+single_tip = containers.load('tiprack-200ul', 'A2', 'Single Tips')
+"""
+Column B
+"""
+multi_tip = containers.load('tiprack-200ul', 'B1', 'Multi Tips')
+"""
+Column C
+"""
+plate96 = containers.load('96-deep-well', 'C1')
+plate384 = containers.load('384-plate', 'C1')
+
+tuberack = containers.load('tube-rack-15_50ml', 'C2')
+
+"""
+Column D
+"""
+liquid_trash = containers.load('trash-box', 'D2', 'Liquid Trash')
+trash = containers.load('trash-box', 'D1', 'Tip Trash')
+"""
+Instruments
+"""
+p300multi = instruments.Pipette(
+    axis='a',
+    name='p300multi',
+    max_volume=300,
+    min_volume=50,
+    channels=8,
+    tip_racks=[multi_tip])
+
+p100single = instruments.Pipette(
+    axis='b',
+    name='p100single',
+    max_volume=100,
+    min_volume=10,
+    channels=1,
+    tip_racks=[single_tip])
+
+"""
+Reagents/variable initialization
+"""
+reagent1 = trough.wells('A1')
+reagent2 = trough.wells('A2')
+reagent3 = trough.wells('A3')
+reagent4 = trough.wells('A4')
+
+
+"""
+Sample Number (Range) for customization: 96-384 (Handled within CSV)
+"""
+volume_example = """
+    50,50,50,50,50,50,50,50
+    50,50,50,50,50,50,50,50
+    50,50,50,50,50,50,50,50
+    50,50,50,50,50,50,50,50
+    50,50,50,50,50,50,50,50
+    50,50,50,50,50,50,50,50
+    50,50,50,50,50,50,50,50
+    50,50,50,50,50,50,50,50
+    50,50,50,50,50,50,50,50
+    50,50,50,50,50,50,50,50
+    50,50,50,50,50,50,50,50
+    50,50,50,50,50,50,50,50
+    """
+
+source_example = """
+    A,B,C
+    D,E,F
+    G,H,
+    I,J,
+    """
+destination_example = """
+    A,,,,,H,
+    ,A,,,,H,
+    ,,,,,H,E
+    ,,,A,,,,
+    ,,,,,,,,
+    ,A,,,,,E
+    ,,G,,,,,
+    ,A,G,E,E
+    ,,,,,,,,
+    ,,G,,,,,
+    ,,G,,,,,
+    ,,,,,,H,
+"""
+
+
+def run_custom_protocol(volumes_csv: FileInput=volume_example,
+                        source_csv: FileInput=source_example,
+                        destination_csv: FileInput=destination_example,
+                        plate_type:
+                            StringSelection('96-flat', '384-flat')='96-flat'):
+
+    """
+    Helper Functions
+    """
+
+    def prewet_tip(pipette):
+        pipette.move_to((reagent1, reagent1.from_center(x=0, y=0, z=-1)))
+
+    def wash(pipette):
+        for _ in range(3):
+            pipette.transfer(
+                pipette.max_volume,
+                reagent2,
+                liquid_trash,
+                new_tip='never')
+
+    """
+    Check which plate you are using
+    """
+    plate = plate384
+    if plate_type == '96-flat':
+        plate = plate96
+
+    """
+    Turns files into dictionaries for volume,
+    sources, destinations respectively
+    """
+    vols = dict()
```

Comment:
It would be helpful for when we come back and read this protocol later if comments explain the shape of these dictionaries -- eg `Create vols dict mapping well number on the destination plate to volume to transfer to that well. The vols dict is of the form: {well_number: volume}`

And similarly for `sources` and `destinations`